cf_xarray/accessor.py

```diff
@@ -2254,18 +2263,29 @@ def decode_vertical_coords(self, prefix="z"):
                     terms["depth_c"] * terms["s"]
                     + (terms["depth"] - terms["depth_c"]) * terms["C"]
                 )
+                # expand dims so ordering is preserved
+                terms["eta"] = terms["eta"].expand_dims(dim={terms["s"].name: terms["s"]}, axis=1)
```

Comment:
I would prefer a more explicit transpose call at the end instead of expanding dimensions. Can you also transpose `romsds` in datasets for the tests.

Reply:
I tried to use the code I have for transposing but it didn't work because some dimensions were identified in the Dataset.dims but not in `cf-xarray` and I wasn't sure how to make it work generically. The code I was trying for that was (originally suggested by you, which I have in `xroms`), if the new depth variable is called `ztemp`:
```
ztemp.cf.transpose(*[dim for dim in ["T", "Z", "Y", "X"] if dim in ztemp.reset_coords(drop=True).cf.axes])
```